qiskit/aqua/algorithms/amplitude_estimators/ae.py

```diff
@@ -351,16 +419,17 @@ def loglikelihood(a):
                 loglik_opt = val
 
         # Convert the value to an estimation
-        val_opt = self.a_factory.value_to_estimation(a_opt)
+        val_opt = self.post_processing(a_opt)
 
         # Store MLE and the MLE mapped to an estimation
         self._ret['ml_value'] = a_opt
         self._ret['mle'] = val_opt
 
     def _run(self) -> 'AmplitudeEstimationResult':
-        # check if A factory has been set
-        if self.a_factory is None:
-            raise AquaError("a_factory must be set!")
+        # check if A factory or state_preparation has been set
+        if self.state_preparation is None:
+            if self.a_factory is None:  # getter emits deprecation warnings, therefore nest
+                raise AquaError('The A operator must be set!')
```

Comment:
Why does this not say that `state_preparation must be set` or `state_preparation (A operator) must be set` if you want to include A operator. In params it calls it a circuit so switching to calling it A operator may not be so obvious to users.
In seeing the AEAlg base class below now I see that is more how state_preparation param is described there (which is a bit different to the text in this class)

Reply:
Good point, In didn't update that after updating the input arguments, thanks 👍

Reply:
Updated in fdda99467